Mathlib/Data/Finsupp/Basic.lean

```diff
@@ -539,6 +539,13 @@ theorem mapDomain_injective {f : α → β} (hf : Function.Injective f) :
   have : mapDomain f v₁ (f a) = mapDomain f v₂ (f a) := by rw [eq]
   rwa [mapDomain_apply hf, mapDomain_apply hf] at this
 
+lemma mapDomain_surjective {f : α → β} (hf : f.Surjective) :
+    (mapDomain (M := M) f).Surjective := by
+  intro x
+  choose y hy using hf
+  use mapDomain y x
+  simp_all [mapDomain, sum_sum_index]
+
```

Comment:
Would you like to split this result as a separate small PR? That will be reviewed and merged quickly. I think I need this result.

By the way, the `choose y hy using hf; use mapDomain y x` may be replaced by `Function.surjInv hf` (untested).

Reply:
Sure! I've split it into #24695.